RecoMuon/TrackerSeedGenerator/plugins/TSGForOIFromL2.cc

```diff
@@ -311,6 +316,93 @@ void TSGForOIFromL2::produce(edm::StreamID sid, edm::Event& iEvent, const edm::E
+      layerCount = 0;
+      for (auto it = tob.rbegin(); it != tob.rend(); ++it) {
+        LogTrace("TSGForOIFromL2") << "TSGForOIFromL2::produce: looping in TOB layer " << layerCount << std::endl;
+        if (useHitLessSeeds_ && hitlessSeedsMadeMuS < maxHitlessSeeds_ && numSeedsMade < maxSeeds_)
+          makeSeedsWithoutHits(**it,
+                               outerTkStateOutside,
+                               *(propagatorOpposite.get()),
+                               estimatorH,
+                               errorSFHitless * SFHld_,
+                               hitlessSeedsMadeMuS,
+                               numSeedsMade,
+                               out);
+        if (hitSeedsMadeMuS < maxHitSeeds_ && numSeedsMade < maxSeeds_)
+          makeSeedsFromHits(**it,
+                            outerTkStateOutside,
+                            *(propagatorOpposite.get()),
+                            estimatorH,
+                            measurementTrackerH,
+                            errorSFHits * SFHd_,
+                            hitSeedsMadeMuS,
+                            numSeedsMade,
+                            layerCount,
+                            out);
+      }
+      LogTrace("TSGForOIFromL2") << "TSGForOIFromL2:::produce: NumSeedsMade = " << numSeedsMade
+                                 << " , layerCount = " << layerCount << std::endl;
+      if (L2muonEta > 0.0) {
+        layerCount = 0;
+        for (auto it = tecPositive.rbegin(); it != tecPositive.rend(); ++it) {
+          LogTrace("TSGForOIFromL2") << "TSGForOIFromL2::produce: looping in TEC+ layer " << layerCount << std::endl;
+          if (useHitLessSeeds_ && hitlessSeedsMadeMuS < maxHitlessSeeds_ && numSeedsMade < maxSeeds_)
+            makeSeedsWithoutHits(**it,
+                                 outerTkStateOutside,
+                                 *(propagatorOpposite.get()),
+                                 estimatorH,
+                                 errorSFHitless * SFHld_,
+                                 hitlessSeedsMadeMuS,
+                                 numSeedsMade,
+                                 out);
+          if (hitSeedsMadeMuS < maxHitSeeds_ && numSeedsMade < maxSeeds_)
+            makeSeedsFromHits(**it,
+                              outerTkStateOutside,
+                              *(propagatorOpposite.get()),
+                              estimatorH,
+                              measurementTrackerH,
+                              errorSFHits * SFHd_,
+                              hitSeedsMadeMuS,
+                              numSeedsMade,
+                              layerCount,
+                              out);
+        }
+        LogTrace("TSGForOIFromL2") << "TSGForOIFromL2:::produce: NumSeedsMade = " << numSeedsMade
+                                   << " , layerCount = " << layerCount << std::endl;
+      }
+
+      if (L2muonEta < 0.0) {
```

Comment:
```suggestion
      else if (L2muonEta < 0.0) {
```
What should happen in the case `L2muonEta==0`?

Reply:
It can enter the first, but I think It doesn't really matter.